Q1. What is the average, median, and standard deviation of house prices?

In [1]:
import pandas as pd

# Replace this list with your actual house price data
house_prices = [350000, 420000, 290000, 850000, 410000, 320000, 550000]

# Create a pandas Series
prices_series = pd.Series(house_prices)

# Calculate metrics
print(f"Average (Mean): ${prices_series.mean():,.2f}")
print(f"Median:         ${prices_series.median():,.2f}")
print(f"Std Deviation:  ${prices_series.std():,.2f}")

Average (Mean): $455,714.29
Median:         $410,000.00
Std Deviation:  $193,550.78


Q2. Which number of bedrooms is most common?

In [2]:
import pandas as pd

# 1. Add your bedroom data here (matching the order of your house prices)
bedrooms = [3, 4, 3, 5, 3, 4, 4]

# 2. Create a DataFrame or Series
bedroom_series = pd.Series(bedrooms)

# 3. Calculate the most common number (Mode)
most_common_bedrooms = bedroom_series.mode()[0]
count = bedroom_series.value_counts().iloc[0]

print(f"The most common number of bedrooms is: {most_common_bedrooms} (appears {count} times)")

The most common number of bedrooms is: 3 (appears 3 times)


Q3. Check if the price data is skewed.

In [3]:
import pandas as pd

# Your house price data
house_prices = [350000, 420000, 290000, 850000, 410000, 320000, 550000]
prices_series = pd.Series(house_prices)

# Calculate skewness
skewness = prices_series.skew()

print(f"Skewness Coefficient: {skewness:.2f}")

# Interpret the result
if skewness > 0.5:
    print("The data is Positively Skewed (Right-skewed) - pulled by high prices.")
elif skewness < -0.5:
    print("The data is Negatively Skewed (Left-skewed) - pulled by low prices.")
else:
    print("The data is relatively Symmetric.")

Skewness Coefficient: 1.73
The data is Positively Skewed (Right-skewed) - pulled by high prices.


Q4. What is the average price for each number of bedrooms?

In [ ]:
import pandas as pd

# Combined data from your previous steps
data = {
    'price': [350000, 420000, 290000, 850000, 410000, 320000, 550000],
    'bedrooms': [3, 4, 3, 5, 3, 4, 4]
}

df = pd.DataFrame(data)

# Group by bedrooms and calculate the average price
avg_price_by_bedroom = df.groupby('bedrooms')['price'].mean().reset_index()

# Format the output for readability
avg_price_by_bedroom['price'] = avg_price_by_bedroom['price'].map('${:,.2f}'.format)
print(avg_price_by_bedroom.to_string(index=False))


Q5.  What is the relationship between living area and price? (Use : Area without Basement)

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# 1. Update your dataset to include the living area (sq ft) for each house
data = {
    'price': [350000, 420000, 290000, 850000, 410000, 320000, 550000],
    'bedrooms': [3, 4, 3, 5, 3, 4, 4],
    'area_no_basement': [1500, 1800, 1200, 3500, 1700, 1400, 2400] # Example area data in sq ft
}

df = pd.DataFrame(data)

# 2. Calculate the Pearson correlation coefficient
correlation = df['area_no_basement'].corr(df['price'])
print(f"Correlation between Living Area and Price: {correlation:.2f}")

# 3. Interpret the relationship
if correlation > 0.7:
    print("There is a strong positive relationship (as area increases, price increases significantly).")
elif correlation > 0.3:
    print("There is a moderate positive relationship.")
else:
    print("There is a weak or no linear relationship.")

# 4. Create a quick scatterplot to visualize it
plt.figure(figsize=(6, 4))
sns.regplot(data=df, x='area_no_basement', y='price', scatter_kws={'s':100}, line_kws={'color':'red'})
plt.title('Living Area vs. House Price')
plt.xlabel('Living Area (Without Basement) sq ft')
plt.ylabel('Price ($)')
plt.grid(True, linestyle='--', alpha=0.6)
plt.show()

Q6. Identify any anomalies where houses have high prices but low areas.

In [ ]:
import pandas as pd

# Data from your previous steps
data = {
    'price': [350000, 420000, 290000, 850000, 410000, 320000, 550000],
    'bedrooms': [3, 4, 3, 5, 3, 4, 4],
    'area_no_basement': [1500, 1800, 1200, 3500, 1700, 1400, 2400]
}

df = pd.DataFrame(data)

# 1. Calculate price per square foot
df['price_per_sqft'] = df['price'] / df['area_no_basement']

# 2. Define an anomaly threshold (e.g., more than 1.5 standard deviations above the average ratio)
mean_ratio = df['price_per_sqft'].mean()
std_ratio = df['price_per_sqft'].std()
threshold = mean_ratio + (1.5 * std_ratio)

anomalies = df[df['price_per_sqft'] > threshold]

# 3. Print results
print(f"Cutoff threshold for high price/low area anomaly: ${threshold:.2f} per sq ft\n")
if anomalies.empty:
    print("No anomalies found! All house prices scale proportionally with their living area.")
else:
    print("Anomalies detected (High price, low area):")
    print(anomalies[['price', 'area_no_basement', 'price_per_sqft']])

Q7. Compare average price based on number of floors and houses with or without waterfront.

In [ ]:
import pandas as pd

# 1. Update the dataset with example values for 'floors' and 'waterfront'
# (Make sure these match your actual dataset rows)
data = {
    'price': [350000, 420000, 290000, 850000, 410000, 320000, 550000],
    'bedrooms': [3, 4, 3, 5, 3, 4, 4],
    'area_no_basement': [1500, 1800, 1200, 3500, 1700, 1400, 2400],
    'floors': [1, 2, 1, 2, 1, 1, 2],                 # Number of floors
    'waterfront': [0, 0, 0, 1, 0, 0, 1]              # 0 = No Waterfront, 1 = Has Waterfront
}

df = pd.DataFrame(data)

# 2. Map waterfront values to readable labels for the final table
df['waterfront'] = df['waterfront'].map({1: 'Waterfront', 0: 'No Waterfront'})

# 3. Create a pivot table showing the average price
pivot_table = df.pivot_table(
    values='price',
    index='floors',
    columns='waterfront',
    aggfunc='mean'
)

# 4. Format numbers to currency style
formatted_pivot = pivot_table.applymap(lambda x: f"${x:,.2f}" if pd.notnull(x) else "N/A")

print("--- Average Price by Floors and Waterfront Views ---")
print(formatted_pivot)

Q8. Identify the minimum and maximum house price. What does this indicate?

In [ ]:
import pandas as pd

# Data from your notebook
house_prices = [350000, 420000, 290000, 850000, 410000, 320000, 550000]
prices_series = pd.Series(house_prices)

min_price = prices_series.min()
max_price = prices_series.max()
price_range = max_price - min_price

print(f"Minimum House Price: ${min_price:,.2f}")
print(f"Maximum House Price: ${max_price:,.2f}")
print(f"Total Price Range:   ${price_range:,.2f}")

Q9. Which location (zipcode/area) has the highest average price?

In [ ]:
import pandas as pd

# 1. Update the dataset to include a 'zipcode' or 'location' column
# (Make sure these match your actual dataset rows)
data = {
    'price': [350000, 420000, 290000, 850000, 410000, 320000, 550000],
    'bedrooms': [3, 4, 3, 5, 3, 4, 4],
    'area_no_basement': [1500, 1800, 1200, 3500, 1700, 1400, 2400],
    'floors': [1, 2, 1, 2, 1, 1, 2],
    'waterfront': [0, 0, 0, 1, 0, 0, 1],
    'zipcode': ['98001', '98002', '98001', '98004', '98002', '98001', '98004'] # Example Zipcodes
}

df = pd.DataFrame(data)

# 2. Group by zipcode and calculate the average price
avg_price_by_zip = df.groupby('zipcode')['price'].mean().reset_index()

# 3. Sort to find the highest average price first
avg_price_by_zip = avg_price_by_zip.sort_values(by='price', ascending=False)

# 4. Format the numbers for clean reading
avg_price_by_zip['price'] = avg_price_by_zip['price'].map('${:,.2f}'.format)

print("--- Average House Price by Zipcode ---")
print(avg_price_by_zip.to_string(index=False))

print(f"\nThe location with the highest average price is Zipcode: {avg_price_by_zip.iloc[0]['zipcode']}")

Q10. Write at least 5 insights from your analysis
Answer
Q10. Write at least 5 insights from your analysis
Based on the complete data structure we have built and analyzed across your previous notebook cells, here are 5 key insights from your housing market analysis:

1. Significant Price Variation and Market Diversity
The dataset exhibits high market diversity, with a total price range of $560,000. House prices span from an entry-level baseline of $290,000 to a premium high of $850,000, indicating that the sample includes multiple tiers of real estate (from affordable starter homes to luxury properties) rather than a uniform neighborhood.

2. High Right-Skewness Driven by a Luxury Outlier
The price data is heavily positively skewed (right-skewed) with a skewness coefficient of 1.73. This is visually and mathematically proven by the fact that the Mean ($455,714.29) is pulled significantly higher than the Median ($410,000.00). The main driver of this skew is the $850,000 property, which acts as a luxury outlier.

3. Extremely Predictable, Linear Price-to-Area Scaling
There is a remarkably strong, consistent relationship between living area (excluding basement) and price. Across all properties, the price per square foot stays fixed in a tight band between $228 and $243. Because of this uniform pricing structure, the dataset contains zero anomalies where a house has a high price paired with low square footage.

4. Direct Premium Associated with Location (Zipcodes)
Location serves as a massive determinant of value in this dataset. Houses in zipcode 98004 command the highest average prices because they house the premium, multi-floor, and waterfront inventory, whereas zipcodes like 98001 and 98002 capture the more affordable, mid-tier segments of the market.

5. Compounding Value Factors (Floors, Bedrooms, and Waterfronts)
Property features scale predictably alongside price. The most common configuration is 3 bedrooms (appearing 3 times). However, as you step up to 4 bedrooms, and ultimately the 5-bedroom configuration, the average price rises sequentially. This space premium compounds heavily when properties feature 2 floors and a Waterfront view, which represents the absolute peak price ceiling in your data.